File: project_environment_all/py_unittest/test_setup_agents.py
import unittest
from environment_all import BombeRLeWorld

class TestBombeRLeWorld(unittest.TestCase):
    def test_setup_agents(self):
        world = BombeRLeWorld()
        agents = [("agent1", True), ("agent2", False)]

        world.setup_agents(agents)

        self.assertEqual(len(world.agents), len(agents))
        self.assertEqual(world.agents[0].agent_dir, agents[0][0])
        self.assertEqual(world.agents[0].name, agents[0][0])
        self.assertEqual(world.agents[0].train, agents[0][1])
        self.assertEqual(world.agents[1].agent_dir, agents[1][0])
        self.assertEqual(world.agents[1].name, agents[1][0])
        self.assertEqual(world.agents[1].train, agents[1][1])

if __name__ == '__main__':
    unittest.main()

File: v0dot2/project_environment_all/js_unittest/test_setup_agents.js
const BombeRLeWorld = require('./environment_all');

class TestBombeRLeWorld {
  test_setup_agents() {
    const world = new BombeRLeWorld();
    const agents = [["agent1", true], ["agent2", false]];

    world.setup_agents(agents);

    assert.equal(world.agents.length, agents.length);
    assert.equal(world.agents[0].agent_dir, agents[0][0]);
    assert.equal(world.agents[0].name, agents[0][0]);
    assert.equal(world.agents[0].train, agents[0][1]);
    assert.equal(world.agents[1].agent_dir, agents[1][0]);
    assert.equal(world.agents[1].name, agents[1][0]);
    assert.equal(world.agents[1].train, agents[1][1]);
  }
}

const test = new TestBombeRLeWorld();
test.test_setup_agents();